### PR TITLE
Reuse the same temporary canvas for creating shader images

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
Instead of creating a new offscreen canvas every time we render the image, just use one shared temporary canvas that is reused whenever needed.

### CODE REVIEW

**Review by kenrussell — COMMENTED** [review]
This is great! I'm only a bystander, but this change looks good to me.

To have it archived here, it's worth mentioning this was related to Google-internal bug ID 214612687.

File: lib/web_ui/lib/src/engine/html/shaders/shader.dart [hunk 1]
```diff
@@ -22,6 +22,21 @@ import 'vertex_shaders.dart';
 const double kFltEpsilon = 1.19209290E-07; // == 1 / (2 ^ 23)
 const double kFltEpsilonSquared = 1.19209290E-07 * 1.19209290E-07;
 
+class TempCanvas {
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/html/shaders/shader.dart:25]: nit: is this really a "temp" canvas? It feel more like a "permanent" canvas :) Maybe call it "SharedCanvas"?

File: lib/web_ui/lib/src/engine/html/shaders/shader.dart [hunk 2]
```diff
@@ -22,6 +22,21 @@ import 'vertex_shaders.dart';
...
+  OffScreenCanvas? _canvas;
+  int currentWidth = 0;
+  int currentHeight = 0;
```
[the remark this answers is not in the document]
Comment (eyebrowsoffire) [reply lib/web_ui/lib/src/engine/html/shaders/shader.dart:28]: They don't need to exist at all actually.

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**